File: Webpack/src/js/leg.ts
import { Destination } from "../js/destination";
import { Origin } from "../js/origin";
export class Leg {
    public Name: string;
    public Type: string;
    public Origin: Origin;
    public Destination: Destination;

    constructor(name: string, type: string) {
        this.Name = name;
        this.Type = type;
    }
}

File: RejseplanAPI/RejseplanAPI/Model/leg.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace RejseplanAPI.Model
{
    public class Leg
    {
        public string Name { get; set; }
        public string Type { get; set; }
        public Origin Origin { get; set; }
        public Destination Destination { get; set; }

        public Leg(string name, string type, Origin origin, Destination destination)
        {
            Name = name;
            Type = type;
            Origin = origin;
            Destination = destination;
        }
    }
}
